dbms/src/Interpreters/Join.h

```diff
@@ -150,6 +151,21 @@ class Join
         RowRefList(const Block * block_, size_t row_num_) : RowRef(block_, row_num_) {}
     };
 
+    /// Map for a time series
+    using ASOFTimeType = UInt32;
+    using AsofGetterType = ColumnsHashing::HashMethodOneNumber<ASOFTimeType, ASOFTimeType, ASOFTimeType, false>;
+    struct TSRowRef
+    {
+        // TODO use the arena allocator to get memory for this
+        // This would require ditching std::map because std::allocator is incompatible with the arena allocator
+        std::map<ASOFTimeType, RowRef> ts;
+
+        TSRowRef() {}
+        void insert(ASOFTimeType t, const Block * block, size_t row_num);
+        std::optional<std::pair<ASOFTimeType, RowRef>> find_asof(ASOFTimeType t) const;
```

Comment:
coding style

Reply:
I ran clang-format with the .clang-format file in the repo on the file and also your own style checking script and neither flagged up anything. Could you elaborate?

Reply:
Quite strange. find_asof is a function, so it should be in camelCase: findAsof

Reply:
Done 817a9244c